[Flaky Test] TestController_RotateCertificates is failing consistently

**Describe the bug**
Even after merging #5187, unit test `TestController_RotateCertificates` is failing consistently. It can be reproduced locally.
```
=== RUN   TestController_RotateCertificates
I0711 10:31:26.786847   71689 ipsec_certificate_controller.go:447] "Shutting down AntreaAgentIPsecCertificateController"
I0711 10:31:26.787848   71689 ipsec_certificate_controller.go:483] "Created new certificate and key for IPSec" cert="/var/folders/9q/0_7t6cs557d_v8mfkhcw5r780000gp/T/antrea-ipsec-test2255709075/fake-node-1-9fba09efc2.crt" key="/var/folders/9q/0_7t6cs557d_v8mfkhcw5r780000gp/T/antrea-ipsec-test2255709075/fake-node-1-9fba09efc2.key"
E0711 10:31:26.789445   71689 ipsec_certificate_controller.go:143] "Error syncing IPSec certificates, requeuing" err="failed to validate new certificate: x509: certificate has expired or is not yet valid: current time 2023-07-11T10:31:26-07:00 is before 2023-07-11T17:31:31Z"
I0711 10:31:27.080704   71689 ipsec_certificate_controller.go:423] "Starting AntreaAgentIPsecCertificateController"
E0711 10:31:27.080748   71689 ipsec_certificate_controller.go:272] "Verifying current certificate configurations failed" err="certificate and key pair is nil"
I0711 10:31:27.080753   71689 ipsec_certificate_controller.go:279] "Start rotating IPsec certificate"
    ipsec_certificate_controller_test.go:388: Sign CSR "fake-node-1-rbv9gsjf" successfully
I0711 10:31:28.280508   71689 ipsec_certificate_controller.go:483] "Created new certificate and key for IPSec" cert="/var/folders/9q/0_7t6cs557d_v8mfkhcw5r780000gp/T/antrea-ipsec-test1123940196/fake-node-1-6f5f574069.crt" key="/var/folders/9q/0_7t6cs557d_v8mfkhcw5r780000gp/T/antrea-ipsec-test1123940196/fake-node-1-6f5f574069.key"
I0711 10:31:28.281264   71689 ipsec_certificate_controller.go:234] "Calculated certificate rotation deadline" expiration="2023-07-11 17:31:36 +0000 UTC" deadline="2023-07-11 17:31:33.118241302 +0000 UTC"
I0711 10:31:28.281283   71689 ipsec_certificate_controller.go:395] "Updating OVS configurations for IPsec certificates"
I0711 10:31:28.281370   71689 ipsec_certificate_controller.go:279] "Start rotating IPsec certificate"
    ipsec_certificate_controller_test.go:388: Sign CSR "fake-node-1-hvlh6q88" successfully
    ipsec_certificate_controller_test.go:315: CSR should not be signed before the rotation deadline
--- FAIL: TestController_RotateCertificates (2.83s)
```

My guess is that the root cause is this error:
```
E0711 10:31:27.080748   71689 ipsec_certificate_controller.go:272] "Verifying current certificate configurations failed" err="certificate and key pair is nil"
```

The error is causing the certificate to be rotated immediately, rather than waiting for the next rotation deadline, hence causing the test to fail because the certificate is rotated too quickly:
https://github.com/antrea-io/antrea/blob/b3aa3a007fd806646293fcf79db457703e9a2cb6/pkg/agent/controller/ipseccertificate/ipsec_certificate_controller.go#L270-L276

--- a/pkg/agent/controller/ipseccertificate/ipsec_certificate_controller.go
+++ b/pkg/agent/controller/ipseccertificate/ipsec_certificate_controller.go
@@ -58,11 +58,12 @@ const (
 	ovsConfigCACertificateKey = "ca_cert"
 	ovsConfigPrivateKeyKey    = "private_key"
 	ovsConfigCertificateKey   = "certificate"
+
+	// certificateWaitTimeout controls the amount of time we wait for certificate approval in
+	// one iteration.
+	certificateWaitTimeout = 15 * time.Minute
 )
 
-// certificateWaitTimeout controls the amount of time we wait for certificate
-// approval in one iteration.
-var certificateWaitTimeout = 15 * time.Minute
 var defaultCertificatesPath = "/var/run/openvswitch"
 
 // Controller is responsible for requesting certificates by CertificateSigningRequest and configure them to OVS
@@ -154,7 +155,7 @@ type certificateKeyPair struct {
 	rotationDeadline time.Time
 }
 
-func (pair *certificateKeyPair) validate() error {
+func (pair *certificateKeyPair) validate(clock clock.Clock) error {
 	if pair == nil {
 		return fmt.Errorf("certificate and key pair is nil")
 	}
@@ -177,6 +178,7 @@ func (pair *certificateKeyPair) validate() error {
 		KeyUsages: []x509.ExtKeyUsage{
 			x509.ExtKeyUsageIPSECTunnel,
 		},
+		CurrentTime: clock.Now(),
 	}
 	if _, err := certificate.Verify(verifyOptions); err != nil {
 		return err
@@ -268,7 +270,7 @@ func (c *Controller) syncConfigurations() error {
 
 	var deadline time.Time
 	// Validate the existing certificate and key pair.
-	if err := c.certificateKeyPair.validate(); err != nil {
+	if err := c.certificateKeyPair.validate(c.clock); err != nil {
 		klog.ErrorS(err, "Verifying current certificate configurations failed")
 		deadline = c.clock.Now()
 	} else {
@@ -281,7 +283,7 @@ func (c *Controller) syncConfigurations() error {
 		if err != nil {
 			return fmt.Errorf("failed to rotate certificate: %w", err)
 		}
-		if err := newCertKeyPair.validate(); err != nil {
+		if err := newCertKeyPair.validate(c.clock); err != nil {
 			newCertKeyPair.cleanup()
 			return fmt.Errorf("failed to validate new certificate: %w", err)
 		}

--- a/pkg/agent/controller/ipseccertificate/ipsec_certificate_controller_test.go
+++ b/pkg/agent/controller/ipseccertificate/ipsec_certificate_controller_test.go
@@ -74,7 +74,6 @@ func newFakeController(t *testing.T, clock clock.WithTicker) *fakeController {
 				csr.ObjectMeta.Name = fmt.Sprintf("%s%s", csr.ObjectMeta.GenerateName, utilrand.String(8))
 				csr.ObjectMeta.GenerateName = ""
 				csr.UID = uuid.NewUUID()
-				csr.CreationTimestamp = metav1.Time{Time: clock.Now()}
 			}
 			return false, csr, nil
 		}),
@@ -270,9 +269,10 @@ func TestController_syncConfigurations(t *testing.T) {
 }
 
 func TestController_RotateCertificates(t *testing.T) {
-	ch := make(chan struct{})
-	defer close(ch)
-	now := time.Now()
+	// It is important to truncate to the second, because the accuracy of notAfter in the
+	// certificate is at the second level. If we don't, the certificate may actually be rotated
+	// before 7s.
+	now := time.Now().Truncate(time.Second)
 	fakeClock := testingclock.NewFakeClock(now)
 	fakeController := newFakeController(t, fakeClock)
 	defer fakeController.mockController.Finish()
@@ -304,7 +304,9 @@ func TestController_RotateCertificates(t *testing.T) {
 	}()
 	fakeController.mockBridgeClient.EXPECT().GetOVSOtherConfig().Times(1)
 	fakeController.mockBridgeClient.EXPECT().UpdateOVSOtherConfig(gomock.Any()).MinTimes(1)
-	go fakeController.Run(ch)
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+	go fakeController.Run(stopCh)
 	<-signCh
 	// the rotation interval is determined by nextRotationDeadline as notBefore + (notAfter -
 	// notBefore) * k, where k is >= 0.7 and <= 0.9. We would therefore expect the rotation
@@ -341,9 +343,16 @@ func newIPsecCertTemplate(t *testing.T, nodeName string, notBefore, notAfter tim
 	}
 }
 
-func createCertificate(t *testing.T, nodeName string, caCert *x509.Certificate, caKey crypto.Signer,
-	publicKey crypto.PublicKey, notBefore time.Time, expirationDuration time.Duration) []byte {
-	template := newIPsecCertTemplate(t, nodeName, notBefore, notBefore.Add(expirationDuration))
+func createCertificate(
+	t *testing.T,
+	nodeName string,
+	caCert *x509.Certificate,
+	caKey crypto.Signer,
+	publicKey crypto.PublicKey,
+	currentTime time.Time,
+	expirationDuration time.Duration,
+) []byte {
+	template := newIPsecCertTemplate(t, nodeName, currentTime, currentTime.Add(expirationDuration))
 	derBytes, err := x509.CreateCertificate(rand.Reader, template, caCert, publicKey, caKey)
 	require.NoError(t, err)
 	certs, err := x509.ParseCertificates(derBytes)
@@ -354,11 +363,13 @@ func createCertificate(t *testing.T, nodeName string, caCert *x509.Certificate, 
 	return encoded
 }
 
-func signCSR(t *testing.T, controller *fakeController,
-	csr *certificatesv1.CertificateSigningRequest, expirationDuration time.Duration) {
+func signCSR(
+	t *testing.T,
+	controller *fakeController,
+	csr *certificatesv1.CertificateSigningRequest,
+	expirationDuration time.Duration,
+) {
 	assert.Empty(t, csr.Status.Certificate)
-	// use the CreationTimestamp as the NotBefore field of issued certificates for testing.
-	assert.False(t, csr.CreationTimestamp.IsZero())
 	block, remain := pem.Decode(csr.Spec.Request)
 	assert.Empty(t, remain)
 	req, err := x509.ParseCertificateRequest(block.Bytes)
@@ -370,7 +381,7 @@ func signCSR(t *testing.T, controller *fakeController,
 	time.Sleep(1 * time.Second)
 
 	newCert := createCertificate(t, req.Subject.CommonName, controller.caCert,
-		controller.caKey, req.PublicKey, csr.CreationTimestamp.Time, expirationDuration)
+		controller.caKey, req.PublicKey, controller.clock.Now(), expirationDuration)
 	toUpdate := csr.DeepCopy()
 	toUpdate.Status.Conditions = append(toUpdate.Status.Conditions, certificatesv1.CertificateSigningRequestCondition{
 		Type:   certificatesv1.CertificateApproved,

